lib/gridfs/grid_store.js

```diff
@@ -73,10 +73,11 @@ const deprecationFn = deprecate(() => {},
  * @param {string} [filename] optional filename for this file, no unique constrain on the field
  * @param {string} mode set the mode for this file.
  * @param {object} [options] Optional settings.
- * @param {(number|string)} [options.w] The write concern.
- * @param {number} [options.wtimeout] The write concern timeout.
- * @param {boolean} [options.j=false] Specify a journal write concern.
- * @param {boolean} [options.fsync=false] Specify a file sync write concern.
+ * @param {(number|string)} [options.w] DEPRECATED: The write concern. Use writeConcern instead.
+ * @param {number} [options.wtimeout] DEPRECATED: The write concern timeout. Use writeConcern instead.
+ * @param {boolean} [options.j=false] DEPRECATED: Specify a journal write concern. Use writeConcern instead.
+ * @param {boolean} [options.fsync=false] DEPRECATED: Specify a file sync write concern. Use writeConcern instead.
```

Comment:
https://docs.mongodb.com/manual/reference/write-concern/index.html
Fsync isn't an option anymore, its equivalent to journal, maybe this is a good opportunity to write a message: "since version x use journal instead"